satpy/writers/cf_writer.py

```diff
@@ -225,18 +226,24 @@ def create_grid_mapping(area):
     return area.area_id, grid_mapping
 
 
-def get_extra_ds(dataset, keys=None):
+def get_extra_ds(dataarray, keys=None):
     """Get the extra datasets associated to *dataset*."""
     ds_collection = {}
-    for ds in dataset.attrs.get('ancillary_variables', []):
-        if keys and ds.name not in keys:
-            keys.append(ds.name)
-            ds_collection.update(get_extra_ds(ds, keys))
-    ds_collection[dataset.attrs['name']] = dataset
-
+    # Retrieve ancillary variable datarrays
```

Comment:
@mraspaud I guess this function can be eliminated since is never called with the `keys` argument. 
And if `keys` is not provided it will never retrieve the ancillary DataArrays right? 
@djhoese what was the purpose of ancillary variables? For what are they used?

Reply:
I did not make this function so I have no idea. I agree that `keys` somehow isn't used except inside this function itself.

Reply:
Looks like the function was added by @mraspaud but the `keys` functionality was added by @zxdawn in https://github.com/pytroll/satpy/pull/1140

Reply:
I guess we can simply remove therefore, since in the current writer I have the `_add_ancillary_variables_attrs` that looks for `ancillary_variables` attribute and replace the eventual list of DataArrays with a list of DataArray names (so netCDF to not complain when writing :D) 
The question is @mraspaud @djhoese .. do we want to write to disk also ancillary variable DataArrays?

Reply:
some data has complementary data included, like quality flags or the like, that what I call ancillary data.